track_and_trace/asset_client/scripts/utils/submit_utils.js

```diff
@@ -0,0 +1,338 @@
+/**
+ * Copyright 2017 Intel Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * ----------------------------------------------------------------------------
+ */
+'use strict'
+
+const _ = require('lodash')
+const sjcl = require('sjcl')
+const request = require('request-promise-native')
```

Comment:
Why the use of this, versus using the existing mithril request function?

Reply:
At this point the script is running before the window object is loaded which is required to use the mithril requests.

Reply:
Ah, that makes sense